sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -153,7 +151,7 @@ export class ServiceClient {
     if (Array.isArray(options.requestPolicyFactories)) {
       requestPolicyFactories = options.requestPolicyFactories;
     } else {
-      let credentialsOrFactory: ServiceClientCredentials | RequestPolicyFactory | undefined = undefined;
+      let credentialsOrFactory: RequestPolicyFactory | undefined = undefined;
       if (isTokenCredential(credentials)) {
```

Comment:
Since what is happening inside this `if` block is workaround to make things work with request policy factory, can we not use `if (!isRequestPolicyFactory(credentials))` instead?